### PR TITLE
issue #160 - fix doctrine/orm joined inheritance anonymization

### PR DESCRIPTION
**Pull request by pounard**

### CODE REVIEW

File: docs/content/anonymization/core-anonymizers.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Core Anonymizers
 
+:::info
+All anonymizers can be configured via attributes on Doctrine ORM entities,
+but inheritance is not fully supported yet,
+[please read this page](doctrine-inheritance) for more information.
```
Comment (SimonMellerin) [inline docs/content/anonymization/core-anonymizers.md:6]: This way is better supported by vitepress:

```suggestion
[please read this page](../doctrine-inheritance) for more information.
```

Comment (SimonMellerin) [reply docs/content/anonymization/core-anonymizers.md:6]: And may be this block should go to `docs/content/anonymization/essentials.md`

File: src/Anonymization/Config/Loader/AttributesLoader.php [hunk 2]
```diff
@@ -51,11 +52,13 @@ public function load(AnonymizationConfig $config): void
 
             $reflexionClass = $metadata->getReflectionClass();
             if ($attributes = $reflexionClass->getAttributes(Anonymize::class)) {
+                // We always only of those attributes, foreach() is required
+                // because of reflection API design.
```
Comment (SimonMellerin) [inline src/Anonymization/Config/Loader/AttributesLoader.php:56]: I think a word is missing?

File: tests/Resources/Loader/TestJoinedChild.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+class TestJoinedChild extends TestJoinedParent
+{
+    #[Anonymize(type: 'constant', options: ['value' => 'https://fr.wikipedia.org/wiki/Wikip%C3%A9dia:Accueil_principal#/media/Fichier:Myotis_crypticus_-_Manuel_Ruedi.jpg'])]
+    #[ORM\Column(length: 255, nullable: true)]
+    private ?string $url = null;
+
+    #[Anonymize(type: 'constant', options: ['value' => 'https://fr.wikipedia.org/wiki/Wikip%C3%A9dia:Accueil_principal#/media/Fichier:Myotis_crypticus_-_Manuel_Ruedi.jpg'])]
+    #[ORM\Column(length: 255, nullable: true)]
+    private ?string $thumbnail_url = null;
+
```
Comment (SimonMellerin) [inline tests/Resources/Loader/TestJoinedChild.php:21]: well, why not :shrug: ?

File: tests/Unit/Anonymization/Configuration/AttributeLoaderTest.php [hunk 4]
```diff
@@ -68,6 +67,43 @@ public function testLoadOk(): void
         self::assertSame('country', $testTableConfig['address_0']->options->get('country'));
     }
 
+    public function testLoadWithJoinedInheritance(): void
+    {
+        $classMetadataFactory = $this->getClassMetadataFactory();
+        $classMetadataFactory->getMetadataFor(TestEntityWithEmbedded::class);
```
Comment (SimonMellerin) [inline tests/Unit/Anonymization/Configuration/AttributeLoaderTest.php:73]: Is it normal to have `TestEntityWithEmbedded` here?

File: docs/content/anonymization/core-anonymizers.md (hunk 1 above)
Comment (pounard) [reply docs/content/anonymization/core-anonymizers.md:6]: The file is in the same folder as the other one, so it seems that my URL is correct.

File: tests/Resources/Loader/TestJoinedChild.php (hunk 3 above)
Comment (pounard) [reply tests/Resources/Loader/TestJoinedChild.php:21]: I copy-pasted the code that was given in the related issue, I just added a foo anonymizer configuration.

File: tests/Unit/Anonymization/Configuration/AttributeLoaderTest.php (hunk 4 above)
Comment (pounard) [reply tests/Unit/Anonymization/Configuration/AttributeLoaderTest.php:73]: I guess it's not, will fix it.

**Pull request merged**